### ISSUE TITLE
Update Pantheon to version 7

### ISSUE THREAD

**Issue by bobby285271** [NixOS/nixpkgs#170361]
Tracking issue for Pantheon 7 (see #121847 for Pantheon 6).

Visit [the project board](https://github.com/NixOS/nixpkgs/projects/21) for lists of random notes that needs attention and [this page](https://bobby285271.github.io/what-changed/) for lists of upstream updates that needs review.

cc @NixOS/pantheon

#### FAQ

> Feel free to add more if you can edit this issue.

##### Can we have Pantheon 7 in NixOS 23.05?

If upstream can then we can.

##### Merge time for #130380 sounds too long for me, can updates happen faster?

Some updates need to be addressed together ([see the project board](https://github.com/NixOS/nixpkgs/projects/21)), I try to split updates into small PRs when the update harmonize with existing parts well. We always merge things when it is ready.

##### How to test a Pantheon update?

For flake users, you can try https://github.com/bobby285271/test-pantheon.

For non-flake users, use:

```
$ env NIX_PATH=nixpkgs=/path/to/nixpkgs:nixos-config=/path/to/configuration.nix nixos-rebuild build-vm
```

Do not forget to adjust the path to the nixpkgs checkout `/path/to/nixpkgs` and the configuration `/path/to/configuration.nix`.

##### Why don't handle the following error for me?

```
Failed assertions:
- You have set services.power-profiles-daemon.enable = true;
which conflicts with services.tlp.enable = true;
```

I am following what is done in GNOME. We have no way of knowing which way a user wants to resolve the conflict.

- To disable tlp: `services.tlp.enable = false;`.
- To disable power-profiles-daemon: `services.power-profiles-daemon.enable = false;`.

##### Why granite 7 is packaged as a separate package?

In short, granite 6 is for GTK3 apps and granite 7 is for GTK4 apps. I don't really have the time and knowledge to port the world to GTK4.

##### Why Mutter and GSD 42?

42 is shipped in ubuntu 22.04 and elementary OS 7. While things build with 43+, bumping mutter can cause regressions on gala and when that happens upstream is not likely going to address it before OS 8. We will default to 43+ only when we are confident enough.

**Comment by bobby285271** [comment]
Blocked updates:

- [ ] [switchboard-plug-keyboard](https://github.com/bobby285271/what-changed#pantheonswitchboard-plug-keyboard-270--head) 3.0.0 - going to address together with gala + gsd updates.
